keyboards/keychron/c3_pro/c3_pro.c

```diff
@@ -61,52 +95,21 @@ bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
 #endif
         case KC_OSSW:
             if (record->event.pressed) {
-                default_layer_xor(1U << 0);
-                default_layer_xor(1U << 2);
-                eeconfig_update_default_layer(default_layer_state);
+                // Switches default layer between `MAC_BASE` and `WIN_BASE` (0 and 2)
+                if (default_layer_state == 0) {
+                    set_single_persistent_default_layer(2);
+                } else if (default_layer_state == 2) {
+                    set_single_persistent_default_layer(0);
+                }
```

Comment:
This logic isnt correct either. `default_layer_state` defaults to `1` (`1<<0`) and would be `4` `(1U << 2)` for layer 2, so will fall through in both cases.